### DISCUSSION TITLE
Return values from LoFTR Matcher

### DISCUSSION THREAD

**Discussion by m0ose** [zju3dv/LoFTR#284] [Q&A]
I am having trouble figuring out what all of the return values are that come from calling
```
batch = {'image0': img0, 'image1': img1}

# Inference with LoFTR and get prediction
with torch.no_grad():
    matcher(batch)
```
Here is a dictionary of them.
```
dict_keys(['image0', 'image1', 'bs', 'hw0_i', 'hw1_i', 'hw0_c', 'hw1_c', 'hw0_f', 'hw1_f', 'conf_matrix', 'b_ids', 'i_ids', 'j_ids', 'gt_mask', 'm_bids', 'mkpts0_c', 'mkpts1_c', 'mconf', 'W', 'expec_f', 'mkpts0_f', 'mkpts1_f'])
```
Specifically what is the difference between `mkpts0_f` and `mkpts0_c`, or what is `hw0_i`?
Is there somewhere where these are described?

PS. I ask because this is such a cool technology and I have a feeling it is capable of more than the demo I ran.

**Comment by tcourat** [comment]
I guess mkpts0_f stands for keypoints at fine scale and mkpts0_c stands for keypoints at coarse scale . So you want to use mkpts0_f and mkpts1_f  .
